app/webpacker/components/Media/MediaTable.jsx

```diff
@@ -0,0 +1,254 @@
+import React, { useEffect, useState } from 'react';
+import _ from 'lodash';
+import { Table, Confirm, Modal, Button } from 'semantic-ui-react';
+import { competitionUrl, apiV0Urls } from '../../lib/requests/routes.js.erb';
+import { countries, continents, years } from '../../lib/wca-data.js.erb';
+import { dateRangeBetween } from '../../lib/helpers/media-table';
+import useLoadedData from '../../lib/hooks/useLoadedData';
+import useSaveAction from '../../lib/hooks/useSaveAction';
+import { DateTime } from 'luxon';
+import {
+  DropdownMenu,
+  DropdownItem,
+  DropdownHeader,
+  DropdownDivider,
+  Dropdown,
+} from 'semantic-ui-react'
+const countryOptions = _.map(countries.byIso2, (country) => ({
+  key: country.iso2,
+  text: country.name,
+  value: country.iso2,
+}));
+const continentOptions = continents.map((continent) => ({
+  key: continent.id,
+  text: continent.name,
+  value: continent.name,
+}));
+const yearOptions = [
+  {
+    key: 'all',
+    text: 'All Years',
+    value: 'All Years',
+  },
+  ...years.map((year) => ({
+    key: year,
+    text: year,
+    value: year,
+  })),
+];
+handleSort = (clickedColumn) => () => {
```

Comment:
This variable should be declared with `const`

Reply:
This has the same name as the dispatch in the `userReducer` below. Also, you can't access these setters outside of the component (plus some of them don't exist). I don't think the reducer has been set up properly (but the intention seems right). I'm happy to help on slack if you want.